### PR TITLE
Added main logic

### PR DESCRIPTION
**Pull request by Eyelessdude**
So, test it

### CODE REVIEW

File: src/test/java/task/dudka/DemoTest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    private static final String TEST_FILE_NAME = "testInput.txt";
+    private static final String WRONG_TEST_FILE_NAME = "aaa.txt";
+
+    @Test
+    public void testMain() {
+        Demo.main(new String[]{TEST_FILE_NAME});
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testMainWrongFileName() {
```
[the remark this answers is not in the document]
Comment (Eyelessdude) [reply src/test/java/task/dudka/DemoTest.java]: Fixed

File: src/main/java/task/dudka/Content.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package task.dudka;
+
+public interface Content {
```
[the remark this answers is not in the document]
Comment (Eyelessdude) [reply src/main/java/task/dudka/Content.java]: Removed to abstract class

File: src/main/java/task/dudka/process/ColumnProcess.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package task.dudka.process;
+
+import task.dudka.Content;
+
+public class ColumnProcess implements Content {
+    private int[][] array;
+    private int columnNumber;
+
+    public ColumnProcess(int[][] matrix, int columnNumber) {
```
[the remark this answers is not in the document]
Comment (Eyelessdude) [reply src/main/java/task/dudka/process/ColumnProcess.java]: Removed to abstract class

File: src/main/java/task/dudka/Demo.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    private static List<String> readListFromFile(File file) {
+        List<String> tempList = new ArrayList<>();
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            String line;
+
+            while ((line = bufferedReader.readLine()) != null) {
+                tempList.add(line);
+            }
+        } catch (IOException e) {
```
[the remark this answers is not in the document]
Comment (Eyelessdude) [reply src/main/java/task/dudka/Demo.java]: Added own exception

File: src/main/java/task/dudka/Demo.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Demo {
```
[the remark this answers is not in the document]
Comment (Eyelessdude) [reply src/main/java/task/dudka/Demo.java]: Done, now it's in FileService class

File: src/main/java/task/dudka/Content.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package task.dudka;
+
+public abstract class Content {
+    protected int[][] matrix;
+    protected int line;
```
Comment (energy2522) [inline src/main/java/task/dudka/Content.java]: Does 'line' as name have any sense?

File: src/main/java/task/dudka/Demo.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        for (int i = 0; i < tempArray[0].length; i++) {
+            contents.add(new ColumnProcess(tempArray, i));
+        }
+
+        for (int i = 0; i < tempArray.length; i++) {
+            contents.add(new RowProcess(tempArray, i));
+        }
+        int max = contents.stream().map(Content::findMaxSequence).max(Integer::compareTo).orElse(0);
+
+        System.out.println(max);
```
Comment (energy2522) [inline src/main/java/task/dudka/Demo.java]: Why do we print it here and then return it?

Comment (Eyelessdude) [reply src/main/java/task/dudka/Demo.java]: Removed

File: src/main/java/task/dudka/Content.java (hunk 6 above)
Comment (Eyelessdude) [reply src/main/java/task/dudka/Content.java]: Renamed it

File: src/test/java/task/dudka/DemoTest.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
+package task.dudka;
+
+import org.junit.Test;
+import task.dudka.exceprion.NonexistentFileException;
+
+public class DemoTest {
+    private static final String TEST_FILE_NAME = "testInput.txt";
+    private static final String WRONG_TEST_FILE_NAME = "aaa.txt";
```
Comment (ardier16) [inline src/test/java/task/dudka/DemoTest.java]: What about `VALID_FILE_NAME` & `INVALID_FILE_NAME`?

File: src/main/java/task/dudka/Demo.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+        }
+
+        for (int i = 0; i < tempArray.length; i++) {
+            contents.add(new RowProcess(tempArray, i));
+        }
+
+        return contents.stream().map(Content::findMaxSequence).max(Integer::compareTo).orElse(0);
+    }
+
+
```
Comment (ardier16) [inline src/main/java/task/dudka/Demo.java]: Extra line break

File: src/main/java/task/dudka/Demo.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+        }
+
+        File file = new File(args[0]);
+
+        int[][] matrixArray = FileService.createArrayFromFile(file);
+
+        System.out.println("Max sequence: " + findMax(matrixArray));
+    }
+
+    private static int findMax(int[][] tempArray) {
```
Comment (ardier16) [inline src/main/java/task/dudka/Demo.java]: What about encapsulating that logic to a separate class?

Main class should have no idea about all that initializations and methods calls

File: src/main/java/task/dudka/Demo.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+        for (int i = 0; i < tempArray[0].length; i++) {
+            contents.add(new ColumnProcess(tempArray, i));
+        }
+
+        for (int i = 0; i < tempArray.length; i++) {
+            contents.add(new RowProcess(tempArray, i));
+        }
+
+        return contents.stream().map(Content::findMaxSequence).max(Integer::compareTo).orElse(0);
```
Comment (ardier16) [inline src/main/java/task/dudka/Demo.java]: All this chainable stuff is very great in taking a bit of space, but it's much harder to keep readability.

At least, I recommend extract parts of logic to separate lines, such as:

```java
return contents.stream()
        .map(Content::findMaxSequence)
        .max(Integer::compareTo)
        .orElse(0)
```

File: src/main/java/task/dudka/FileService.java [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+
+        int[][] matrixArray = list.stream().map(row -> row.split(SPLITTER)).map(array -> {
+            int[] intArray = new int[array.length];
+
+            for (int i = 0; i < array.length; i++) {
+                intArray[i] = Integer.valueOf(array[i]);
+            }
+
+            return intArray;
+        }).collect(Collectors.toList()).toArray(new int[list.size()][]);
```
Comment (ardier16) [inline src/main/java/task/dudka/FileService.java]: The same thing about chaining. It's easy to miss `.collect(Collectors.toList()).toArray(new int[list.size()][])` part

File: src/test/java/task/dudka/exception/NonexistentFileExceptionTest.java [hunk 13]
```diff
@@ -0,0 +1,28 @@
+package task.dudka.exception;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import task.dudka.exceprion.NonexistentFileException;
+
+public class NonexistentFileExceptionTest {
```
Comment (ardier16) [inline src/test/java/task/dudka/exception/NonexistentFileExceptionTest.java]: Do we really need it? We have built-in `FileNotFoundExcepion` class that can be used IMO

If we do, maybe let's rename it to `FileNotExistsException`? @energy2522, what do you think?

**Pull request closed without merging**